File: 卡码/代码随想录/JS/贪心/03-困难/LeetCode_0435.js
//435. 无重叠区间
const eraseOverlapIntervals = function(intervals) {
	if(intervals.length==0) return 0;
	intervals.sort((a,b)=>a[0]-b[0]);
	let res=0;
	for(let i=1;i<intervals.length;i++){
		if(intervals[i][0]<intervals[i-1][1]){
			res++;
			intervals[i][1]=Math.min(intervals[i][1],intervals[i-1][1])
		}
	}
	return res;
};

File: 卡码/代码随想录/Go/贪心/03-困难/LeetCode_0435.go
// 435. 无重叠区间
package main

import (
	"math"
	"sort"
)

func eraseOverlapIntervals(intervals [][]int) int {
	if len(intervals)==0 {
		return 0
	}
	sort.Slice(intervals,func (i,j int) bool{
		return intervals[i][0]<intervals[j][0]
	})
	res:=0
	for i:=1;i<len(intervals);i++ {
		if intervals[i][0]<intervals[i-1][1] {
			res++
			intervals[i][1]=int(math.Min(float64(intervals[i][1]),float64(intervals[i-1][1])))
		}
	}
	return res
}
